boat_simulator/nodes/physics_engine/kinematics_computation.py

```diff
@@ -0,0 +1,100 @@
+"""This module contains the kinematics computations for the boat."""
+
+from typing import Tuple
+from numpy.typing import ArrayLike
+from boat_simulator.common import utils
+from boat_simulator.common.types import Scalar
+from boat_simulator.nodes.physics_engine.kinematics_data import KinematicsData
+import numpy as np
+import boat_simulator.common.constants as constants
+
+
+class BoatKinematics:
+    # Private class member default
+    __timestep = 1.0
+    __relative_data = KinematicsData()
+    __global_data = KinematicsData()
+
+    def __init__(self, timestep: Scalar) -> None:
+        self.__timestep = timestep
+
+    # Updates the kinematics data and returns the kinematics data objects
+    def step(self, net_force: ArrayLike, net_torque: ArrayLike
+             ) -> Tuple[KinematicsData, KinematicsData]:
+
+        next_relative_acceleration = self.__compute_next_relative_acceleration(
+            self.boat_mass, net_force
+        )
+        next_relative_velocity = self.__compute_next_relative_velocity(
+            self.relative_data.cartesian_velocity, next_relative_acceleration
+        )
```

Comment:
You want to use the current acceleration, not the next acceleration. Most of the function calls for updating that follow are experiencing the same issue (i.e. using the information from the next timestep $t+1$ when we really want to use the information at the current timestep $t$).

```python
next_relative_velocity = self.__compute_next_relative_velocity(
    self.relative_data.cartesian_velocity, self.relative_data.cartesian_acceleration
)
```